libraries/AP_Camera/AP_Camera_MAVLinkCamV2.cpp

```diff
@@ -47,6 +47,10 @@ bool AP_Camera_MAVLinkCamV2::record_video(bool start_recording)
         return false;
     }
 
+    if ((start_recording & is_recording) == 1U) {
+        return false;
+    }
```

Comment:
I think you should let this one past, blocking this means that if the camera recording ever gets of of sync with what AP is expecting we can't fix it.